example/android/javalib/3-linting/build.mill

```diff
@@ -0,0 +1,108 @@
+// == Linting with Basic Config
+
+// This Mill build configuration includes a linting step, which is essential for ensuring code
+// quality and adhering to best practices in Android development projects. Running the `androidLintRun` task
+// produces a detailed HTML report by default, identifying potential issues in the code, such as performance,
+// security, and usability warnings. This helps maintain the health and quality of the codebase.
+
+// In this example, we set a custom config to generate the report file in txt format and not HTML.
+
+//// SNIPPET:BUILD
+package build
+
+import mill._
+import mill.javalib.android.{AndroidSdkModule, AndroidAppModule, AndroidLintReportFormat}
+
+// Create and configure an Android SDK module to manage Android SDK paths and tools.
+object androidSdkModule0 extends AndroidSdkModule {
+  def buildToolsVersion = "35.0.0"
+  def bundleToolVersion = "1.17.2"
+}
+
+// Actual android application with linting config
+object app extends AndroidAppModule {
+  def androidSdkModule = mill.define.ModuleRef(androidSdkModule0)
+
+  // Set path to the custom `lint.xml` config file. It is usually at the root of the project
+  def androidLintConfigPath = Task { Some(PathRef(millSourcePath / "lint.xml")) }
+
+  // Set path to the custom `baseline.xml` file. It is usually at the root of the project
+  def androidLintBaselinePath = Task { Some(PathRef(millSourcePath / "lint-baseline.xml")) }
+
+  // Set the linting report to be generated as a text file
+  def androidLintReportFmt = Task { AndroidLintReportFormat.Txt }
+
+  // Set to true (default), stops the build if errors are found
+  def abortOnError = false
+
+  // Specify additional flag options for Android Lint not defined in `AndroidLintModule`
+  def androidLintArgs: T[Seq[String]] = T {
+
+    // Set path to generated `.jar` files in this case (or `.class` files)
+    val cp = runClasspath().map(_.path).filter(os.exists).mkString(":")
+
+    // Set path to the location of the project source codes
+    val src = sources().map(_.path).filter(os.exists).mkString(":")
+    Seq(
+      "--classpath",
+      cp,
+      "--sources",
+      src
+    )
```

Comment:
I think this should be moved to the `AndroidAppModule`, this will be the same for 99% of setups and end-user should have everything working out of the box with the reasonable defaults.